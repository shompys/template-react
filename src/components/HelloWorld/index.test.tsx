import { describe, it, expect, vi } from "vitest";
import { render } from "@testing-library/react";

import { HelloWorld } from ".";

const mockData = {
    title: "Hello {{ name }}",
};

vi.mock("react-i18next", () => ({
    useTranslation: () => {
        return {
            t: (str: string, obj: Record<string, string> | undefined = {}) => {
                const replace = (value = "", replace: Record<string, string> | undefined = {}) =>
                    value.replace(/\{\{(.*?)\}\}/g, (_, token) => {
                        const tokenIndex = token.trim();

                        return replace[tokenIndex] || tokenIndex;
                    });

                if (obj) {
                    return replace(mockData[str as keyof typeof mockData], obj);
                }

                return mockData[str as keyof typeof mockData];
            },
        };
    },
}));

describe("<HelloWorld />", () => {
    it("Should render App component", async () => {
        const { container } = render(<HelloWorld />);

        expect(container).toMatchSnapshot();
    });
});
