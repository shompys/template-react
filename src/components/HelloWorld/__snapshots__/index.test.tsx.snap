// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<HelloWorld /> > Should render App component 1`] = `
<div>
  <h1
    class="text-white"
  >
    Hello pepe
  </h1>
</div>
`;
